docs/design/Education-organization-Endpoints.md

```diff
@@ -0,0 +1,153 @@
+# Education Organizations Endpoints
+
+## Overview
+
+Provides a consolidated view of education organizations across all Ed-Fi ODS
+instances through REST API endpoints. The data is refreshed on a scheduled
+basis.
+
+## Features
+
+* **REST API Endpoints:**
+  * `GET /v2/educationorganizations` or `GET /v1/educationorganizations` -
+    Returns all education organizations from all instances
+  * `GET /v2/educationorganizations/{instanceId}` or
+    `GET /v1/educationorganizations/{instanceId}` - Returns education organizations
+    for a specific instance
+
+* **Automatic Data Refresh:**
+  * Quartz.NET scheduled job runs every 6 hours by default
+  * No manual refresh endpoint - data is refreshed automatically via background
+    job
+
+* **Cross-Database Support:**
+  * Works with both SQL Server and PostgreSQL
+  * Uses C# service layer for dynamic database connections and efficient data
+    refresh
+
+## Database Schema
+
+### Tables
+
+* `adminapi.EducationOrganizations` - Stores consolidated education organization
+  data
+
+  ```sql
+  CREATE TABLE [adminapi].[EducationOrganizations] (
+        [Id] INT IDENTITY(1,1) NOT NULL,
+        [InstanceId] INT NOT NULL,
+        [InstanceName] NVARCHAR(100) NOT NULL,
+        [EducationOrganizationId] INT NOT NULL,
+        [NameOfInstitution] NVARCHAR(75) NOT NULL,
+        [ShortNameOfInstitution] NVARCHAR(75) NULL,
+        [Discriminator] NVARCHAR(128) NOT NULL,
+        [ParentId] INT NULL,
+        [OdsDatabaseName] NVARCHAR(255) NULL,
+        [LastRefreshed] DATETIME2 NOT NULL DEFAULT GETUTCDATE(),
+        [LastModifiedDate] DATETIME2 NULL,
+        CONSTRAINT [PK_EducationOrganizations] PRIMARY KEY ([Id])
+    );
+  ```
+
+## API Usage Examples
+
+### Get All Education Organizations
+
+```http
+GET /v2/educationorganizations
+Authorization: Bearer <token>
+```
+
+### Get Education Organizations for Specific Instance
+
+```http
+GET /v2/educationorganizations/123
+Authorization: Bearer <token>
+```
+
+## Configuration
+
+### Quartz.NET Job Scheduling
+
+Add to `appsettings.json`:
+
+```json
+{
+   "AppSettings": {
+   "EducationOrgsRefreshIntervalInHours": 6
```

Comment:
If health check's refresh interval is in minutes, then perhaps this one should be in minutes as well.

---

I wonder about creating a section for refresh intervals. Something like:

```
"backgroundProcessIntervals": {
  "educationOrganizations": 360,
  "healthCheck": 10,
  "instances": 30
}
```

Actually, I'm not sure if instance management will be on a schedule. So, creating a whole section for just two things seems like overkill.